From d50fdb5998b26b3cc41384acfc15a3ccd8d35a87 Mon Sep 17 00:00:00 2001
From: Vadim Shtayura <vadimsh@chromium.org>
Date: Thu, 24 Jun 2021 18:30:51 -0700
Subject: Recognize sigs.k8s.io domains.

Apply a hack to treat "/vN" as part of the package name.
---
 util/normalizename_test.go | 5 +++++
 util/util.go               | 4 ++++
 2 files changed, 9 insertions(+)

diff --git a/util/normalizename_test.go b/util/normalizename_test.go
index df1b59f..da93efd 100644
--- a/util/normalizename_test.go
+++ b/util/normalizename_test.go
@@ -20,6 +20,11 @@ func TestNormalizeName(t *testing.T) {
 			root:  "github.com/dgraph-io/badger/v3",
 			extra: "y",
 		},
+		{
+			input: "sigs.k8s.io/structured-merge-diff/v3/v",
+			root:  "sigs.k8s.io/structured-merge-diff/v3",
+			extra: "v",
+		},
 		{
 			input: `github.com\Masterminds\cookoo\web\io\foo`,
 			root:  "github.com/Masterminds/cookoo",
diff --git a/util/util.go b/util/util.go
index 5e92e3a..8a6dddf 100644
--- a/util/util.go
+++ b/util/util.go
@@ -228,6 +228,10 @@ var vcsList = []*vcsInfo{
 		host:    "hub.jazz.net",
 		pattern: `^(?P<rootpkg>hub\.jazz\.net/git/[a-z0-9]+/[A-Za-z0-9_.\-]+)(/[A-Za-z0-9_.\-]+)*$`,
 	},
+	{
+		host:    "sigs.k8s.io",
+		pattern: `^(?P<rootpkg>sigs\.k8s\.io/[A-Za-z0-9_.\-]+(/v[0-9]+)?)(/[A-Za-z0-9_.\-]+)*$`,
+	},
 	{
 		host:    "go.googlesource.com",
 		pattern: `^(?P<rootpkg>go\.googlesource\.com/[A-Za-z0-9_.\-]+/?)$`,
-- 
2.32.0.93.g670b81a890-goog

